src/construct_CTPG.jl

```diff
@@ -0,0 +1,69 @@
+function CTPG_train(dynamics_plant::Function, dynamics_controller::Function, cost_running::Function, cost_terminal::Function, cost_regularisor::Function, policy_NN, scenario; 
+    solve_alg = Tsit5(), sense_alg = InterpolatingAdjoint(autojacvec = ZygoteVJP()), ensemble_alg = EnsembleThreads(), opt_1 = ADAM(0.01), opt_2 = LBFGS(), progress_plot = true, solve_kwargs...)
```

Comment:
여기에 있는 `sense_alg` 나 `ensemble_alg` 등의 default key 는 없어도 될 것 같은데, 어떻게 생각하시나요?
(아마 DiffEq.jl 에서 기본값을 지원할 것 같은데...)

대신 관련 kwargs 를 묶어서 받고, line 39 쯤에 해당 kwargs를 넣는 식으로 하면 더 좋을 것 같습니다.

+) suggestion 처럼, docstring 을 만들어서 인자들 및 해당 함수의 기능에 대한 설명을 넣으면 좋을 것 같아요. [blank] 에 해당
```suggestion
"""
    CTPG_train(dynamics_plant::Function, dynamics_controller::Function, cost_running::Function, cost_terminal::Function, cost_regularisor::Function, policy_NN, scenario; 
    solve_alg = Tsit5(), sense_alg = InterpolatingAdjoint(autojacvec = ZygoteVJP()), ensemble_alg = EnsembleThreads(), opt_1 = ADAM(0.01), opt_2 = LBFGS(), progress_plot = true, solve_kwargs...)

[blank]
"""
function CTPG_train(dynamics_plant::Function, dynamics_controller::Function, cost_running::Function, cost_terminal::Function, cost_regularisor::Function, policy_NN, scenario; 
    solve_alg = Tsit5(), sense_alg = InterpolatingAdjoint(autojacvec = ZygoteVJP()), ensemble_alg = EnsembleThreads(), opt_1 = ADAM(0.01), opt_2 = LBFGS(), progress_plot = true, solve_kwargs...)
```

Reply:
DiffEqFlux에서 기본값을 지원하기는 할텐데요, 
default key로 관심사인 제어 문제에 알맞는 알고리즘을 명시적으로 부여해주는 편이 낫지 않을까 판단하였습니다.

sensitivity analysis 알고리즘, 최적화 알고리즘 옵션 선택에 따라 안정성과 속도 차이가 있고 하다보니 뭔가 kwarg로 받는 게 좋을 것 같기는 한데....

묶어서 받는 편은 고민해보겠습니다.

Reply:
그럼 "제어 문제에 알맞는 알고리즘을 명시적으로 부여했음"을 docstring 에 추가해주시겠어요?

그럼 나중에 추적이 더 쉬울 것 같습니다 ㅎㅎ

Reply:
추천해준대로 docstring 추가하였습니다